src/http/server.clj

```diff
@@ -293,6 +294,24 @@
          (log/infof "\"%s %s HTTP/1.1\" %s %s" method'  uri status len))
        response))))
 
+(defn pw-prompt []
+  (str (do (println "Password: ") (.readPassword (System/console)))))
+
+(defn maybe-inject-auth
+  [auth handler]
+  (fn [req]
+    (handler (if auth
+               (assoc-in req [:headers "authorization"] auth)
+               req))))
+
+(defn parse-auth [auth]
+  "make sure pw is present, if not prompt for it"
```

Comment:
It's better to avoid shortcuts like "pw" in docstrings. And start from a capital letter.

Reply:
Agree - also should I consider making it `(-defn parse-auth [auth] ...)` since it's really internal?